### PR TITLE
feat(sdk): Allow to hide columns in CollectionBrowser

### LINKED ISSUE #49232: Ability to hide "Last edited at/by" on the Collection Browser component

### PR DESCRIPTION
**Pull request by deniskaber**
Closes https://github.com/metabase/metabase/issues/49232

### Description

This exposes a prop to specify visible columns for CollectionBrowser.

### How to verify

1. Run SDK storybook
2. Check CollectionBrowser > Without Some Columns story

### Demo

https://github.com/user-attachments/assets/ee19aefc-0a0b-4f79-be41-f3f24b9b27fa

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: enterprise/frontend/src/embedding-sdk/components/public/CollectionBrowser/CollectionBrowser.tsx [hunk 1]
```diff
@@ -23,6 +23,19 @@ const USER_FACING_ENTITY_NAMES = [
...
+  | "name"
+  | "lastEditedBy"
+  | "lastEditedAt";
+
+const COLLECTION_BROWSER_LIST_COLUMNS: CollectionBrowserListColumns[] = [
+  "type",
+  "name",
+  "lastEditedBy",
+  "lastEditedAt",
+];
```
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/components/public/CollectionBrowser/CollectionBrowser.tsx:37]: tip: you could also do the reverse to generate the type from the array. I noticed that the above "UserFacingEntityName" type already does this exact thing.

```suggestion
const COLLECTION_BROWSER_LIST_COLUMNS= [
  "type",
  "name",
  "lastEditedBy",
  "lastEditedAt",
] as const;

type CollectionBrowserListColumns = (typeof COLLECTION_BROWSER_LIST_COLUMNS)[number]
```

File: enterprise/frontend/src/embedding-sdk/components/public/CollectionBrowser/CollectionBrowser.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  return <CollectionBrowser {...args} />;
+};
+
+export const Default = {
+  render: Template,
+
+  args: {},
+};
+
+export const WithoutSomeColumns = {
```
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/components/public/CollectionBrowser/CollectionBrowser.stories.tsx:26]: nit: this story name is fine, but I'd name it `WithTypeAndNameColumn` just to be explicit.

File: frontend/src/metabase/collections/components/CollectionContent/constants.ts [hunk 3]
```diff
@@ -1,6 +1,30 @@
...
+  "lastEditedBy",
+  "lastEditedAt",
+  "actionMenu",
+] as const;
+
+export type CollectionContentTableColumn =
+  (typeof COLLECTION_CONTENT_COLUMNS)[number];
+
+export type CollectionContentTableColumnsMap = {
+  [key in CollectionContentTableColumn]: true;
```
Comment (heypoom) [inline frontend/src/metabase/collections/components/CollectionContent/constants.ts:17]: Is it possible to specify `false` as the key? (e.g. `{actionMenu: false}`) - just curious why the type requires `true` here

File: frontend/src/metabase/components/ItemsTable/utils.ts [hunk 4]
```diff
@@ -12,3 +16,11 @@ export const getContainerQuery = (props: ResponsiveProps) =>
...
       }) { display: none; }`
     : "";
+
+export const getVisibleColumnsMap = (
+  visibleColumns: CollectionContentTableColumn[],
+) =>
+  visibleColumns.reduce((result, item) => {
+    result[item] = true;
+    return result;
+  }, {} as CollectionContentTableColumnsMap);
```
Comment (heypoom) [inline frontend/src/metabase/components/ItemsTable/utils.ts:26]: I wonder if we can use `Object.fromEntries` here

```tsx
Object.fromEntries(visibleColumns.map(column => [column, true]))
```

File: frontend/src/metabase/components/ItemsTable/BaseItemTableRow.tsx [hunk 5]
```diff
@@ -65,7 +65,7 @@ export const TableRow = ({
       onCopy={onCopy}
       onMove={onMove}
       onToggleSelected={onToggleSelected}
-      showActionMenu={showActionMenu}
+      visibleColumnsMap={visibleColumnsMap}
```
Comment (heypoom) [inline frontend/src/metabase/components/ItemsTable/BaseItemTableRow.tsx:68]: nit: I noticed that the other component uses `visibleColumns` while this prop uses `visibleColumnsMap`, perhaps we want to use the same prop name?

**Review by heypoom — COMMENTED** [review]
Nice! Added some feedback in the comments

File: frontend/src/metabase/collections/components/CollectionContent/types.ts
Comment (heypoom) [inline frontend/src/metabase/collections/components/CollectionContent/types.ts:1]: PS. looks like `types.ts` is now an empty file

File: frontend/src/metabase/collections/components/CollectionContent/constants.ts (hunk 3 above)
Comment (heypoom) [reply frontend/src/metabase/collections/components/CollectionContent/constants.ts:17]: @deniskaber Ah, I just realized this is an internal data structure - that makes total sense. My bad.

File: frontend/src/metabase/components/ItemsTable/BaseItemTableRow.tsx (hunk 5 above)
Comment (heypoom) [reply frontend/src/metabase/components/ItemsTable/BaseItemTableRow.tsx:68]: @deniskaber I totally missed that the type differs there, thanks!

**Review by heypoom — APPROVED** [review]
I tested this locally in Storybook and this looks good 👍🏻

**Review by npretto — APPROVED** [review]
Tested locally and works as expected with autocompletion 🚀

**Pull request merged**